shell/server/installer.js

```diff
@@ -41,6 +47,15 @@ Meteor.methods({
 });
 
 deletePackage = function (packageId) {
+  // Mark package for possible deletion;
+  Packages.update({_id: packageId}, {$set: {shouldCleanup: true}});
+}
+
+var deletePackageInternal = function (package) {
+  verifyIsMainReplica();
+
+  var packageId = package._id;
+
   if (packageId in installers) {
     return;
```

Comment:
I suppose it might make sense to unset `shouldCleanup` here, rather than in `AppInstaller.done()` per my previous suggestion.